nninit.lua

```diff
@@ -270,4 +270,95 @@ nninit.sparse = function(module, tensor, sparsity)
   return module
 end
 
+--[[
+--  Aghajanyan, A. (2017)
+--  Convolution Aware Initialization
+--  arXiv preprint arXiv:1702.06295
+--]]
+nninit.convaware = function(module, tensor, options)
+  -- The author of the paper provided a reference implementation for Keras: https://github.com/farizrahman4u/keras-contrib/pull/60
+
+  local sizes = tensor:size()
+  if #sizes ~= 4 then
+    error("nninit.convaware only supports 2d convolutions, feel free to issue a pull request to extend this implementation")
+  end
+
+  -- Make sure that the signal library is available, which provides the Fourier transform
+  local requireSignal = function()
+    require('signal')
+  end
+  if pcall(requireSignal) == false then
+    error("nninit.convaware requires the signal library, please make sure to install it: https://github.com/soumith/torch-signal")
+  end
+
+  -- Store the sizes of the convolution tensor to make the implementation easier to read
+  local filterCount = sizes[1]
+  local filterStacks = sizes[2]
+  local filterRows = sizes[3]
+  local filterCols = sizes[4]
+
+  -- Due to the irfft2 interface of the signal library, we currently have to restrict the filter size
+  if filterRows ~= filterCols then
+    error("nninit.convaware requires the filters to have the same number of rows and columns, feel free to issue a pull request to extend this implementation")
+  end
+
+  -- Calculate "fanIn" and "fanOut" for 2d convolution tensors based on module conventions
+  local fanIn = filterStacks * filterRows * filterCols
+  local fanOut = filterCount * filterRows * filterCols
+
+  -- Setup options where "std" specifies the noise to break symmetry in the inverse Fourier transform
+  options = options or {}
+  std = options.std or 0.05
+
+  -- Specify the variables for the frequency domain tensor
+  local fourierTensor = require('signal').rfft2(torch.zeros(filterRows, filterCols))
```

Comment:
By loading `signal` once at the top, `require('signal')` can be replaced neatly.